javascript/code-challenges/challenges-03.test.js

```diff
@@ -0,0 +1,303 @@
+'use strict';
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 1 - Review
+
+Write a function called addTwo that takes in an array and adds two to every value using a for loop. Place the new value in a new array. Return the new array.
+------------------------------------------------------------------------------------------------ */
+
+const addTwo = (arr) => {
+  const output = [];
+  for(let num of arr) {
+    output.push(num + 2);
+  }
+  return output;
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 2
+
+Write a function named typeNum that, given an array as input, uses filter to return an array containing only the numbers.
+
+For example, typeNum([1, 'bob' ,3]) returns [1,3].
+------------------------------------------------------------------------------------------------ */
+
+const typeNum = (arr) => {
+  return arr.filter(any => typeof any === 'number');
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 3
+
+Write a function named containsAnd that, given an array of strings as input, uses filter to return an array containing only strings that contain 'and' within the string.
+
+For example, containsAnd(['panda', 'ran', 'and']) returns ['panda', 'and'].
+------------------------------------------------------------------------------------------------ */
+
+const containsAnd = (arr) => {
+  return arr.filter(str => str.includes('and'));
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 4
+
+Write a function named oddValues that, given an array of integers as input, uses filter to return an array containing only the odd integers.
+
+For example, oddValues([1,2,3]) returns [1,3].
+------------------------------------------------------------------------------------------------ */
+
+const oddValues = (arr) => {
+  return arr.filter(num => num % 2);
```

Comment:
this is actually really clever. Given that "0" is a falsey value, if a number is even it will equate to false and therefore be filtered out. Very cool.